add concept paragraph

abigaildelarosa opened the pull request:
Hi @sampsyo, here is another pull request with the concept paragraph at the end of the getting started chapter.  About the standard programming, I will check what can I improve in that file.

abigaildelarosa commented:
this is related to #2475

sampsyo commented:
Thanks for continuing to pursue this! Now that I see the new text here at the end, I think it's much more "in context"—it flows from the earlier text that says "there's much more to learn about beets," etc.

On the other hand, there's clearly a lot more to say about these topics. What do you think about starting a new documentation page, as #2277 originally suggested? We could then give a little preview here and link to the new page.

wordofglass commented on docs/guides/main.rst:
```diff
@@ -290,9 +290,19 @@ You can always get help using the ``beet help`` command. The plain ``beet help``
 import`` gives more specific help about the ``import`` command.
 
+Core Concepts
+-------------
```
I think this should be moved before the *Keep Playing* section.

wordofglass commented on docs/guides/main.rst:
```diff
@@ -290,9 +290,19 @@ You can always get help using the ``beet help`` command. The plain ``beet help``
+Core Concepts
+-------------
+
+* *Library*: Is the beets database model only meant to catalog users music , which means that it only represents links to music files. The library is commposed by @items(songs) and @albums. The metadata album is shared by the track groups. Each item belongs to an album, otherwise is a @singleton. Users can import songs with the ``beet import`` command. 
```
The sentence about sharing metadata isn't very clear IMO (I'm not sure I understood your intention with that sentence correctly, the following points might be misguided):
* I guess you meant *album metadata* instead of  *metadata album*?
* The term *track groups* is not explained, something along the lines *items it contains* would stick to the expression introduced in the preceding sentence.

wordofglass commented on docs/guides/main.rst:
```diff
@@ -290,9 +290,19 @@ You can always get help using the ``beet help`` command. The plain ``beet help``
+* *Library*: Is the beets database model only meant to catalog users music , which means that it only represents links to music files. The library is commposed by @items(songs) and @albums. The metadata album is shared by the track groups. Each item belongs to an album, otherwise is a @singleton. Users can import songs with the ``beet import`` command. 
+* *Queries*: Represent the data subsets that the user wants to retrieve from the library. 
+* *Paths and Destinations*: Each item has a @destination, which is the path it would reside at. In its default setup, the beet import command copies items to their destination as they are imported. 
+* *Auto-Tagging*: Beets used two auto-tagging workflows, one for Albums and one for Items (i.e., singletons). This ones retrieve metadata provider by MusicBrainz, Beatport etc.
```
used -> uses
In both cases, metadata is retrieved from providers such as MusicBrainz.

abigaildelarosa commented:
hi @sampsyo, what do you think about the last update? I still thinking whether I should move part of the introduction paragraph on the beets workflow section to the very beginning of the getting started guide or leave it as it is? For my personal point of view, it makes sense to first describe install procedure and then introduction to the beets workflow. Afterwards explain technical configurations.

sampsyo commented on docs/guides/main.rst:
```diff
@@ -125,6 +125,25 @@ trouble or you have more detail to contribute here, please direct it to
+In this section we introduce a briefly description of the beets workflows, which might help you to 
+understand the basics behind the program and the key commands. In case that you want
+to contribute as a programmer, make sense about how to navigate through the beets
+components and give better structure to your propose future plugins.
```
A suggestion for a simpler way to say this:

> Beets relies on a few core concepts that underly every feature and every plugin you'll use. Understanding these common pieces can help you see how to harness their full power.

sampsyo commented on docs/guides/main.rst:
```diff
@@ -125,6 +125,25 @@ trouble or you have more detail to contribute here, please direct it to
+
+* **Paths and Destinations**. Each *Item* has a *destination*, which is the path where it would reside at if it were renamed according to the user's preferred scheme. In its default setup, the ``beet import`` command copies items to their destination as they are imported. The destination consist of two parts: *root directory* (specified as ``directory`` in the config file)  and the template-speci ...
+
+* **Auto-Tagging**. There are two *auto-tagging* workflows: one for Album and one for item. The album tagger is selected by default and uses the filesystem to group files into Albums. Afterwards the information is used to query metadata provided by MusickBrainz for canonical metadata.
```
Hmm… I'm not sure if this belongs here. The distinction between a singleton and an album is probably useful to know, but it's not specific to auto-tagging. And in some ways, I don't think auto-tagging is a "core concept" in beets—it's one feature, but it's not an underlying source of its power. :smiley:

sampsyo commented on docs/guides/main.rst:
```diff
@@ -125,6 +125,25 @@ trouble or you have more detail to contribute here, please direct it to
+
+In case that additional information is needed, please do not forget to take a look at our `Wiki`_.
+
+.. _Wiki : https://github.com/beetbox/beets/wiki
```
Should we really link the wiki? Most of the stuff there is about project management and development…

sampsyo commented on docs/guides/main.rst:
```diff
@@ -125,6 +125,25 @@ trouble or you have more detail to contribute here, please direct it to
+to contribute as a programmer, make sense about how to navigate through the beets
+components and give better structure to your propose future plugins.
+
+* **Data Model**. The beets *Library* is a database that catalogs your music, which does not contain any music files; as a database, it saves the local links to your music files. The *Library* is composed by *Items* (songs) and *Albums*. The Album is a sequence of *Items* and consist of a set of album-level metadata that is share among the *Items*. Each *Item* belongs to an *Album*, songs without ...
```
"local links" -> "paths" (i.e., the filenames)

sampsyo replied:
Some simplified language:

> An Album is a collection of Items, and it also contains its own, separate album-level metadata that is shared among all of its tracks. Some Items don't have an album; these are called *singletons*.

sampsyo replied:
Should the discussion about `beet import` really go here? We're about to introduce the commands, but this section (if I understand correctly) is about concepts more than about specific commands.

sampsyo commented on docs/guides/main.rst:
```diff
@@ -125,6 +125,25 @@ trouble or you have more detail to contribute here, please direct it to
+
+* **Data Model**. The beets *Library* is a database that catalogs your music, which does not contain any music files; as a database, it saves the local links to your music files. The *Library* is composed by *Items* (songs) and *Albums*. The Album is a sequence of *Items* and consist of a set of album-level metadata that is share among the *Items*. Each *Item* belongs to an *Album*, songs without ...
+
+* **Queries**. This are used to select subsets of *Albums* and *Items*. They can be compared with the SQL statement ``SELECT``. Their main goal is to retrieve your data subsets. Almost all the commands in beets perform first a query task ( ``list``, ``remove``, ``stats``, ``modify``, ``move`` etc.) except for the per-Album or per-Item operations.
```
> In beets, you use *queries* to select Albums and Items.

sampsyo replied:
I recommend not mentioning SQL here, which be confusing to non-programmers.

sampsyo replied:
What are those per-Album and per-Item operations? I might leave this part of the sentence off—it suffices to just say:

> Nearly every beets command takes a query as an argument that tells it what music to operate on.

sampsyo commented on docs/guides/main.rst:
```diff
@@ -125,6 +125,25 @@ trouble or you have more detail to contribute here, please direct it to
+
+* **Queries**. This are used to select subsets of *Albums* and *Items*. They can be compared with the SQL statement ``SELECT``. Their main goal is to retrieve your data subsets. Almost all the commands in beets perform first a query task ( ``list``, ``remove``, ``stats``, ``modify``, ``move`` etc.) except for the per-Album or per-Item operations.
+
+* **Paths and Destinations**. Each *Item* has a *destination*, which is the path where it would reside at if it were renamed according to the user's preferred scheme. In its default setup, the ``beet import`` command copies items to their destination as they are imported. The destination consist of two parts: *root directory* (specified as ``directory`` in the config file)  and the template-speci ...
```
Perhaps link to the section of the configuration where we describe path formats when they're introduced?

sampsyo replied:
There's a missing space in "destination.The"

sampsyo commented on docs/guides/main.rst:
```diff
@@ -296,3 +315,5 @@ Please let me know what you think of beets via `the discussion board`_ or
 .. _the discussion board: http://discourse.beets.io
 .. _twitter: http://twitter.com/b33ts
+
+
```
This looks unintentional. :smiley:

sampsyo commented on docs/guides/main.rst:
```diff
@@ -1,8 +1,8 @@
-Welcome to `beets`_! This guide will help you begin using it to make your music
-collection better.
+Welcome to `beets`_! This guide will help you begin using it to make your 
+music collection better. 
```
No need for this change. :smiley:

abigaildelarosa commented:
Hi @sampsyo, I changed the text based on your inputs. Let me know what do you think about the draft 2. Below there is a list of the changes. 

-change introduction text.

Data Model section

- change local links to local paths/
- Simplified language about the singletons.
- Delete import command description given that we want to describe only the concepts.

Queries section

-Simplified first sentence.
-Delete SQL statement given that non programmer will not understand it.
-Delete per Album and per Item sentence.

Paths and destinations.

add path link
fix missing space.

Delete Auto-Tagging section

-Remove wiki link

sampsyo commented:
This is looking great! Thank you again for continuing to work on it!

I'd love to hear opinions from other users too about how helpful this looks and what else they'd like to see. I'll post a link to our Twitter feed and see if we get any comments…

sampsyo commented:
I also pushed a few changes. Please take a look!

Kraymer commented on docs/guides/main.rst:
```diff
@@ -290,6 +290,36 @@ You can always get help using the ``beet help`` command. The plain ``beet help``
+
+* **The library.** The beets *Library* is a database that catalogs your music.
+  The Library does not contain any music files itself;
+because it's a database, it saves the paths (i.e., the filenames)
```
indent

sampsyo replied:
Right; as pointed out by Travis: https://travis-ci.org/beetbox/beets/jobs/213252579#L885

(This may be my fault.)

Kraymer commented on docs/guides/main.rst:
```diff
@@ -290,6 +290,36 @@ You can always get help using the ``beet help`` command. The plain ``beet help``
+  The Library does not contain any music files itself;
+because it's a database, it saves the paths (i.e., the filenames)
+  to your music files. The Library is composed of *Items* (songs) and *Albums*.
+  The Album is a collection of Items, and it also contains its own, separate
```
The Album => An Album ?

jtpavlock commented:
I'm going to echo @sampsyo's statement in saying I think this belong's in it's own doc page, and perhaps referenced in the 'Keep Playing' section. Also, putting it in it's own doc page could allow you to get more in depth and perhaps replace the github wiki entirely.
